aioredis/pool.py

```diff
@@ -160,6 +161,8 @@ def closed(self):
     @asyncio.coroutine
     def wait_closed(self):
         """Wait until pool gets closed."""
+        if self._close_waiter is None:
+            self._close_waiter = async_task(self._do_close(), loop=self._loop)
```

Comment:
I don't get your point there.

If `close` is called first, it will create the `self._close_waiter` task

If `wait_closed` is called first, it will create the self._close_waiter` and then wait until `close_state` is set